core/cas-server-core-util/src/main/java/org/apereo/cas/util/cipher/BaseBinaryCipherExecutor.java

```diff
@@ -59,12 +61,26 @@ public BaseBinaryCipherExecutor(final String encryptionSecretKey,
         if (StringUtils.isBlank(encryptionSecretKey)) {
             LOGGER.warn("Secret key for encryption is not defined under [{}]. CAS will attempt to auto-generate the encryption key",
                     getEncryptionKeySetting());
-            this.encryptionSecretKey = RandomStringUtils.randomAlphabetic(encryptionKeySize);
+            final String key = (new Base64RandomStringGenerator(encryptionKeySize)).getNewString();
             LOGGER.warn("Generated encryption key [{}] of size [{}]. The generated key MUST be added to CAS settings under setting [{}].",
-                    this.encryptionSecretKey, encryptionKeySize, getEncryptionKeySetting());
+                key, encryptionKeySize, getEncryptionKeySetting());
+            encryptionKey = EncodingUtils.decodeBase64(key);
         } else {
-            this.encryptionSecretKey = encryptionSecretKey;
+            final boolean base64 = EncodingUtils.isBase64(encryptionSecretKey);
+            byte[] key = new byte[0];
+            if (base64) {
+                key = EncodingUtils.decodeBase64(encryptionSecretKey);
+            }
+            if (base64 && key.length == encryptionKeySize) {
```

Comment:
The old keys may be valid base64 due to the characters used. If it's base64 and the key size is not the same as the decoded length, either it's an old key or misconfigured.